### PR TITLE
Better error messaging with hamcrest matchers

### PR DESCRIPTION
**Pull request by peternied**
### Description
Better error messaging with hamcrest matchers, see examples of how the error messages look inline.

### Check List
- [X] New functionality includes testing
  - [X] All tests pass, including unit test, integration test and doctest
- [ ] ~New functionality has been documented~
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: RFS/src/test/java/com/rfs/common/RestClientTest.java [hunk 1]
```diff
@@ -38,51 +42,39 @@ public void testGetEmitsInstrumentation() throws Exception{
...
-                .collect(Collectors.toList());
-        Assertions.assertTrue(!finishedSpans.isEmpty());
-
-        try {
-            Assertions.assertEquals(String.join("\n", List.of("httpRequest", "httpRequest", "createSnapshot")),
-                    finishedSpans.stream().map(SpanData::getName).collect(Collectors.joining("\n")));
-        } catch (Throwable e) {
-            log.error(finishedSpans.stream().map(Object::toString).collect(Collectors.joining("\n")));
-            throw e;
+            assertThat("Checking bytes {send, read} for context '" + kvp.getKey() + "'", new long[]{bytesSent, bytesRead}, equalTo(kvp.getValue()));
```
Comment (peternied) [inline RFS/src/test/java/com/rfs/common/RestClientTest.java:60]: On failure looks like:
```
RestClientTest > testGetEmitsInstrumentation() FAILED
    java.lang.AssertionError: Checking bytes {send, read} for context 'createGetSnapshotContext'
    Expected: [<1313L>, <66L>]
         but: was [<133L>, <66L>]
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at com.rfs.common.RestClientTest.testGetEmitsInstrumentation(RestClientTest.java:60)
```

File: RFS/src/test/java/com/rfs/common/RestClientTest.java [hunk 2]
```diff
@@ -38,51 +42,39 @@ public void testGetEmitsInstrumentation() throws Exception{
...
         }
 
+        final var finishedSpans = rootContext.instrumentationBundle.getFinishedSpans();
+        final var finishedSpanNames = finishedSpans.stream().map(SpanData::getName).collect(toList());
+        assertThat(finishedSpanNames, containsInAnyOrder("httpRequest", "httpRequest", "createSnapshot"));
```
Comment (peternied) [inline RFS/src/test/java/com/rfs/common/RestClientTest.java:65]: On failure looks like
```
RestClientTest > testGetEmitsInstrumentation() FAILED
    java.lang.AssertionError: 
    Expected: iterable with items ["httpRequ1est", "httpRequest", "createSnapshot"] in any order
         but: not matched: "httpRequest"
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
        at com.rfs.common.RestClientTest.testGetEmitsInstrumentation(RestClientTest.java:65)
```

File: RFS/src/test/java/com/rfs/common/RestClientTest.java [hunk 3]
```diff
@@ -38,51 +42,39 @@ public void testGetEmitsInstrumentation() throws Exception{
...
                 new long[]{133,66})) {
-            var span = finishedSpans.get(i++);
-            Assertions.assertEquals(span.getAttributes().get(RfsContexts.GenericRequestContext.BYTES_SENT_ATTR),
-                    counts[0]);
-            Assertions.assertEquals(span.getAttributes().get(RfsContexts.GenericRequestContext.BYTES_READ_ATTR),
-                    counts[1]);
+            var span = httpRequestSpansByTime.get(i++);
+            long bytesSent = span.getAttributes().get(RfsContexts.GenericRequestContext.BYTES_SENT_ATTR);
+            long bytesRead = span.getAttributes().get(RfsContexts.GenericRequestContext.BYTES_READ_ATTR);
+            assertThat("Checking bytes {send, read} for httpRequest " + i, new long[]{bytesSent, bytesRead}, equalTo(expectedBytes));
```
Comment (peternied) [inline RFS/src/test/java/com/rfs/common/RestClientTest.java:77]: On failure looks like
```
RestClientTest > testGetEmitsInstrumentation() FAILED
    java.lang.AssertionError: Checking bytes {send, read} for httpRequest 1
    Expected: [<1139L>, <66L>]
         but: was [<139L>, <66L>]
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        at com.rfs.common.RestClientTest.testGetEmitsInstrumentation(RestClientTest.java:77)
```

File: RFS/src/test/java/com/rfs/common/RestClientTest.java [hunk 4]
```diff
@@ -18,9 +20,11 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
-import static org.junit.jupiter.api.Assertions.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+import static java.util.stream.Collectors.toList;
```
Comment (gregschohn) [inline RFS/src/test/java/com/rfs/common/RestClientTest.java:26]: No static imports please.  The previous one shouldn't have been there either.

**Review by gregschohn — APPROVED** [review]
Will merge this, but I'm going to cleanup the static imports

File: RFS/src/test/java/com/rfs/common/RestClientTest.java (hunk 4 above)
Comment (peternied) [reply RFS/src/test/java/com/rfs/common/RestClientTest.java:26]: Hmm, maybe I can sell you on some static imports - but I'll consider that my challenge to sell you on the value.  Until we have a style enforcement lets table back and forth on this.

Comment (peternied) [reply RFS/src/test/java/com/rfs/common/RestClientTest.java:26]: (So I am absolutely fine with you removing them for your changes)

**Pull request merged**